modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/edit_folder.jsp

```diff
@@ -488,34 +490,42 @@ renderResponse.setTitle(headerTitle);
 
 			searchContainer.deleteRow(tr, link.getAttribute('data-rowId'));
 
-			A.one(
-				'#<portlet:namespace />defaultFileEntryTypeId-' +
+			var option = document.getElementById(
+				'<portlet:namespace />defaultFileEntryTypeId-' +
 					link.getAttribute('data-rowId')
-			).remove();
+			);
+
+			option.parentElement.removeChild(option);
 
 			<portlet:namespace />documentTypesChanged = true;
```

Comment:
This is the possible bug that I mentioned [here](https://github.com/liferay-frontend/liferay-portal/pull/11#issuecomment-647447036).

Before this PR, `<portlet:namespace />documentTypesChanged` was a global and now it is not, because we had:

```js
<aui:script>
    var <portlet:namespace />documentTypesChanged = false;
```

and we changed it to:

```js
<aui:script use="liferay-search-container">
    var <portlet:namespace />documentTypesChanged = false;
```

ie. it is now in an IIFE due to the `use` (horrible leaky abstraction, I know), which means that is is local to the IIFE.

But down here on line 496, this line is now broken. Note that this is in a *separate* `<aui:script use="liferay-search-container">` block, which means that it is its own closure. It cannot "see" the `<portlet:namespace />documentTypesChanged` in the other closure, and there is no `var` declaration here so this one *is* setting the variable on the window.

TL;DR: Both of these should be:

```js
window['<portlet:namespace />documentTypesChanged'] = ...;
```

Reply:
Or, now that I look at them... just merge the two script tags together. It looks like they are two identical consecutive script tags:

```js
<aui:script use="liferay-search-container">
    // stuff
</aui:script>

<aui:script use="liferay-search-container">
    // more stuff
</aui:script>
```

They could just be:

```js<aui:script use="liferay-search-container">
    // stuff
    // more stuff
</aui:script>
```

I could have sworn I commented on a similar thing in a PR over the last few days, but it must have been another one... 🤔

Reply:
Ah, it was [this one](https://github.com/wincent/liferay-portal/pull/344#pullrequestreview-429073581).